llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java

```diff
@@ -417,15 +419,21 @@ public OrcTail getOrcTailFromCache(Path path, Configuration jobConf, CacheTag ta
   }
 
   @Override
-  public RecordReader<NullWritable, VectorizedRowBatch> llapVectorizedOrcReaderForPath(Object fileKey, Path path, CacheTag tag, List<Integer> tableIncludedCols,
-      JobConf conf, long offset, long length) throws IOException {
+  public RecordReader<NullWritable, VectorizedRowBatch> llapVectorizedOrcReaderForPath(Object fileKey, Path path,
+      CacheTag tag, List<Integer> tableIncludedCols, JobConf conf, long offset, long length, Reporter reporter)
+      throws IOException {
 
-    OrcTail tail = getOrcTailFromCache(path, conf, tag, fileKey);
+    OrcTail tail = null;
+    if (tag != null) {
```

Comment:
Why not put this inside the `getOrcTailFromCache`?

Reply:
- getOrcTailFromCache should always be called with a non-null CacheTag instance.
- to produce a record reader in llapVectorizedOrcReaderForPath we either know the tag or not depending the source of invocation
- - in case tag is known: we can do the metadata lookup and attach the result to the OrcSplit
- - in case it's unknown: we create the split without the tail info, and let OrcEncodedDataReader find out the metadata later